app/controllers/rhapsody_realities.js

```diff
@@ -80,10 +80,73 @@ class RhapsodyController {
                 res.render('admin_list_rhapsody', { viewData, layout: 'admin_main' });
             })
             .catch(() => {
-                res.flash('error', 'There was an error retrieving rhapsodies, Please try again');
+                req.flash('error', 'There was an error retrieving rhapsodies, Please try again');
                 res.render('admin_add_rhapsody', { viewData, layout: 'admin_main' });
             });
     }
+
+    /**
+    * rhapsody list
+    *
+    * @param req
+    * @param res
+    */
+    editView(req, res, next) {
+        let viewData = {
+            menuActive: 'rhapsody'
+        };
+        let rhapsodyId = req.params.id;
+        let error = req.query.e;
+        if(error){
+            viewData.error = decodeURIComponent(error);
+        }
+
+        if (!rhapsodyId) {
+            req.flash('error', 'No Rhapsody ID specified');
+            res.redirect('/admin/list-rhapsody-realities');
+        }
+        this.rhapsodyService.getRhapsody(rhapsodyId)
+            .then(rhapsody => {
+                viewData.rhapsody = rhapsody;
+                res.render('admin_edit_rhapsody', { viewData, layout: 'admin_main' });
+            })
+            .catch(() => {
+                req.flash('error', 'An unknown error has occurred, Please try again later');
+                res.redirect('/admin/list-rhapsody-realities');
+            });
+    }
+
+    /**
+     * rhapsody list
+     *
+     * @param req
+     * @param res
+     */
+    update(req, res, next) {
+        let viewData = {
+            menuActive: 'rhapsody'
+        };
+        let body = req.body;
+        // Validate Form Input
+        let error = this.rhapsodyService.validateRhapsodyData(req);
+        if (error) {
+            this.logger.error('Validation Error: ', error);
+            let errorMessages = error.map(item => item.msg);
+            let url = '/admin/edit-rhapsody-realities/' + body.id + '?e=' + encodeURIComponent(error[0].msg);
```

Comment:
its not ideal to pass the error via url
thats why we have flash configured.

change line 137 below to render the same page, and the error will be show to the user like we do in saving

make sure you have {{{printMessages}}) on the view , cos thats the placeholder where the flash message will be delivered